saleor/payment/gateways/paypal/__init__.py

```diff
@@ -0,0 +1,140 @@
+import decimal
+import logging
+
+from paypalcheckoutsdk.orders import OrdersCaptureRequest, OrdersCreateRequest
+from paypalcheckoutsdk.payments import CapturesRefundRequest
+
+from ... import TransactionKind
+from ...interface import GatewayConfig, GatewayResponse, PaymentData
+from .utils import get_paypal_client
+
+logger = logging.getLogger(__name__)
+
+
+"""
+Paypal API references:
+https://developer.paypal.com/docs/checkout/
+https://github.com/paypal/Checkout-Python-SDK
+https://developer.paypal.com/docs/api/orders/v2/
+https://medium.com/paypal-engineering/launch-v2-paypal-checkout-apis-45435398b987
+"""
+
+
+def get_client_token(**_):
+    """Not implemented for paypal gateway currently.
+
+    The client token can be generated in the client.
+    """
+
+
+def get_paypal_order_id(config: GatewayConfig, amount: float, currency: str) -> str:
+    """Only for use in tests.
+
+    to get the token (Paypal order id), when payment has not yet been created.
+    """
+    client = get_paypal_client(**config.connection_params)
+    request = OrdersCreateRequest()
+    request.prefer("return=representation")
+    request.request_body(
+        {
+            "intent": "CAPTURE",
+            "purchase_units": [
+                {"amount": {"currency_code": currency, "value": amount}}
+            ],
+        }
+    )
+
+    response = client.execute(request)
+    return response.result.id
+
+
+def authorize(
+    payment_information: PaymentData, config: GatewayConfig
+) -> GatewayResponse:
+
+    # transaction_kind =\
+    #     TransactionKind.CAPTURE if config.auto_capture else TransactionKind.AUTH
+
+    client = get_paypal_client(**config.connection_params)
+    request = OrdersCaptureRequest(payment_information.token)
+    try:
+        response = client.execute(request)
+    except IOError as ioe:
+        # if isinstance(ioe, HttpError):
+        #     # Something went wrong server-side
+        #     print(ioe.status_code)
+        error_message = getattr(ioe, "status_code", repr(ioe))
+        return GatewayResponse(
+            is_success=False,
+            action_required=False,
+            amount=payment_information.amount,
+            transaction_id=payment_information.token,
+            kind=TransactionKind.CAPTURE,
```

Comment:
Loose thought, but from what I see this implementation only supports "capture" payments without prior authorization and manual capture? Maybe we could then move this entire implementation to the `capture` method below?

Reply:
I am just travelling around opened PR... this one is very solid! 
@maarcingebala Ain't you worry only about logic?... i found this code could be improved a little, partially...